### PR TITLE
PR fixing Issue 15: Sudoers and NMAP

### PR DESCRIPTION
**Pull request by mosesrenegade**
Under a developers workstation that is not docker running native debian or ubuntu the developer may have a non root account in which intrigue runs under. NMAP will not work correctly as many of the options will require root privileges. Current work around is to have an entry in the sudoers file to not prompt for a password (NOPASSWD: ALL). System will now check for UID 0 and if not it will insert the sudo command ahead of nmap.

This also adds the .ruby-gemset option for rbenv.

### CODE REVIEW

File: lib/tasks/nmap_scan.rb [hunk 1]
```diff
@@ -47,7 +47,11 @@ def run
       _log "Scanning #{scan_item} and storing in #{temp_file}"
       _log "NMap options: #{nmap_options}"
 
-      nmap_string = "nmap #{scan_item} #{nmap_options} -sSUV -P0 --top-ports 25 -O --max-os-tries 2 -oX #{temp_file}"
+      if Process.uid == 0
```
Comment (mosesrenegade) [inline lib/tasks/nmap_scan.rb]: I'm not 100% sure if this is the best option to check if your running the process as root, i'm just guessing that this is a way to check. I think it needs a test case for this.

Comment (jcran) [inline lib/tasks/nmap_scan.rb]: This works. May want to check EID too, but good for now. https://ruby-doc.org/core-2.1.0/Process/UID.html

**Comment by jcran** [comment]
Looks good. Merged, thanks!

**Comment by jcran** [comment]
Reverted this out to merge into develop first

**Pull request merged**